### ISSUE TITLE
[Bug] zos_job_submit edge cases don't always return duration

### ISSUE THREAD

**Issue by richp405** [ansible-collections/ibm_zos_core#1032]
### Is there an existing issue for this?

- [X] There are no existing issues.

### Are the dependencies a supported version?

- [X] The dependencies are supported.

### IBM Z Open Automation Utilities

v1.2.5

### IBM Enterprise Python

v3.12.x

### IBM z/OS Ansible core Version

v1.7.0

### ansible-version

v2.14.x

### z/OS version

v2.5

### Ansible module

zos_job_submit

### Bug description

zos_job_submit usually returns 'duration' as a member of a job run.  During a recent update, unit testing showed an edge case where duration was not being returned.  This shows in def test_job_submit_max_rc(ansible_zos_module, args), first case.

________________________ test_job_submit_max_rc[args0] _________________________
           for result in results.contacted.values():
                # Should fail normally as a non-zero RC will result in job submit failure
                if args["max_rc"] is None:
                    assert result.get("msg") is not None
                    assert result.get('changed') is False
                    # On busy systems, it is possible that the duration even for a job with a non-zero return code
                    # will take considerable time to obtain the job log and thus you could see either error msg below
                    #Expecting: - "The job return code 8 was non-zero in the job output, this job has failed"
                    #           - Consider using module zos_job_query to poll for a long running job or
                    #             increase option \\'wait_times_s` to a value greater than 10.",
>                   if result.get('duration') >= args["wait_time_s"]:
E                   TypeError: '>=' not supported between instances of 'NoneType' and 'int'

So, in an error with a return code, there are some (at least 1) case where duration is not defined (NoneType).

There were several paths that need to be audit/traced to make sure the return consistently contains this value.  In the mean time, the following stop-gap code resolves the issue for unit testing:

                if result.get('duration'):
                    duration = result.get('duration')
                else:
                    duration = 0

                if duration >= args["wait_time_s"]

This forces 'duration' to always be defined, even if result.get(duration) is not defined.

I did not see impact to similar job_submit-based playbooks, so I believe this is a borderline/edge case test.  This needs to be fixed for the longer-term good of the product.

### Playbook verbosity output.

Found in unit testing that includes test_zos_job_submit_func.py testing, 

specifically def test_job_submit_max_rc(ansible_zos_module, args):

### Ansible configuration.

```YAML
# Was found by pipeline, using Ansible 2.14
[defaults]
forks = 25
timeout = 30
collections_paths = ~/.ansible/collections
module_utils = ~/.ansible/collections/ansible_collections/ibm/ibm_zos_core/plugins/module_utils

[ssh_connection]
pipelining = True
ssh_args = -o ControlMaster=auto -o ControlPersist=28800 -o ConnectTimeout=240s

[connection]
pipelining = True
```

### Contents of the inventory

```YAML
# Ran on a single EC (01132) with standard config and super-shares.
host: ec01132a.vmec.svl.ibm.com
user: omvsadm
python_path: /allpython/3.9/usr/lpp/IBM/cyp/v3r9/pyz/bin/python3
environment:
    _BPXK_AUTOCVT: "ON"
    _CEE_RUNOPTS: "'FILETAG(AUTOCVT,AUTOTAG) POSIX(ON)'"
    _TAG_REDIR_ERR: txt
    _TAG_REDIR_IN: txt
    _TAG_REDIR_OUT: txt
    LANG: C
    ZOAU_HOME: "/zoau/v1.2.5"
    ZOAU: "/zoau/v1.2.5/bin"
    ZOAU_ROOT: "/zoau/v1.2.5"
    LIBPATH: "/zoau/v1.2.5/lib:/allpython/3.9/usr/lpp/IBM/cyp/v3r9/pyz/lib:/lib:/usr/lib:."
    ZOAUTIL_DIR: "/zoau/v1.2.5"
    PYTHONPATH: "/zoau/v1.2.5/lib"
    PYTHON_HOME: "/allpython/3.9/usr/lpp/IBM/cyp/v3r9/pyz"
    PYTHON: "/allpython/3.9/usr/lpp/IBM/cyp/v3r9/pyz/bin"
    PATH: "/zoau/v1.2.5/bin:/allpython/3.9/usr/lpp/IBM/cyp/v3r9/pyz/bin:/bin:/usr/sbin:/var/bin:/usr/lpp/java/java180/J8.0_64/bin"
    PYTHONSTDINENCODING: "cp1047"
```

### Contents of `group_vars` or `host_vars`

```YAML
(from extended inventory file)
```

**Comment by ddimatos** [comment]
@richp405 - I don't think this solves the issue but due note that during the release of 1.9.0 as I was making some job submission edits to support TYPRUN, I noticed the duration was not correct, there was some variable passing by reference not value and it threw the duration off by about 5 seconds. 

In those code changes, I corrected this [line](https://github.com/ansible-collections/ibm_zos_core/blob/73eae1c3cd1786b242b359d5cf58d84d5b5eaf96/plugins/modules/zos_job_submit.py#L665) 
```
        if job_output is not None:
            duration = job_output_txt[0].get("duration") if not None else duration
```

I also am confused to as how this could not be zero given in the branch you are working from , duration is initialized at this point:
https://github.com/ansible-collections/ibm_zos_core/blob/73eae1c3cd1786b242b359d5cf58d84d5b5eaf96/plugins/modules/zos_job_submit.py#L890

Makes me think this was a result of a module failure in the test case resulting in null values.

**Comment by richp405** [comment]
@ddimatos Thanks for pointing that line out.  It made me view the issue differently.

I think I've figured out what was happening:  In a few cases, while **duration** (local value) is set, I'm seeing 2 cases where the value is **NOT** assigned back to results.  I'll check that trace today and close it or correct it, based on what I find.

**Comment by ddimatos** [comment]
For what its worth, I merged in the 1.9 changes to `dev` and ran this test locally with the safety set of `duration=0` and was not able to recreate. 

I ran twice, I did not check my change in which removes the safety check which we probably should do to surface any module issues, anyhow running locally twice with no issues.

```
(venv-2.16) ./ac --ac-test --host ec33017a --python 3.10 --zoau 1.3.0 --file tests/functional/modules/test_zos_job_submit_func.py --test test_job_submit_max_rc  --debug true
=========================================================================================== test session starts ===========================================================================================
platform darwin -- Python 3.11.5, pytest-8.0.0, pluggy-1.4.0
ansible: 2.16.3
rootdir: /Users/ddimatos/git/gh/ibm_zos_core/tests
configfile: pytest.ini
plugins: ansible-24.1.2, mock-3.12.0
collected 3 items

tests/functional/modules/test_zos_job_submit_func.py ...

============================================================================================ warnings summary =============================================================================================
functional/modules/test_zos_job_submit_func.py::test_job_submit_max_rc[args2]
functional/modules/test_zos_job_submit_func.py::test_job_submit_max_rc[args2]
functional/modules/test_zos_job_submit_func.py::test_job_submit_max_rc[args1]
functional/modules/test_zos_job_submit_func.py::test_job_submit_max_rc[args0]
functional/modules/test_zos_job_submit_func.py::test_job_submit_max_rc[args0]
  /Users/ddimatos/git/gh/ibm_zos_core/venv/venv-2.16/lib/python3.11/site-packages/ansible/plugins/loader.py:1479: UserWarning: AnsibleCollectionFinder has already been configured
    warnings.warn('AnsibleCollectionFinder has already been configured')

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
================================================================================ 3 passed, 5 warnings in 81.00s (0:01:20) =================================================================================
(venv-2.16) ./ac --ac-test --host ec33017a --python 3.10 --zoau 1.3.0 --file tests/functional/modules/test_zos_job_submit_func.py --test test_job_submit_max_rc  --debug true
=========================================================================================== test session starts ===========================================================================================
platform darwin -- Python 3.11.5, pytest-8.0.0, pluggy-1.4.0
ansible: 2.16.3
rootdir: /Users/ddimatos/git/gh/ibm_zos_core/tests
configfile: pytest.ini
plugins: ansible-24.1.2, mock-3.12.0
collected 3 items

tests/functional/modules/test_zos_job_submit_func.py ...

============================================================================================ warnings summary =============================================================================================
functional/modules/test_zos_job_submit_func.py::test_job_submit_max_rc[args2]
functional/modules/test_zos_job_submit_func.py::test_job_submit_max_rc[args2]
functional/modules/test_zos_job_submit_func.py::test_job_submit_max_rc[args1]
functional/modules/test_zos_job_submit_func.py::test_job_submit_max_rc[args0]
functional/modules/test_zos_job_submit_func.py::test_job_submit_max_rc[args0]
  /Users/ddimatos/git/gh/ibm_zos_core/venv/venv-2.16/lib/python3.11/site-packages/ansible/plugins/loader.py:1479: UserWarning: AnsibleCollectionFinder has already been configured
    warnings.warn('AnsibleCollectionFinder has already been configured')

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
================================================================================ 3 passed, 5 warnings in 70.10s (0:01:10) =================================================================================
(venv-2.16)
```


<img width="877" alt="image" src="https://github.com/ansible-collections/ibm_zos_core/assets/25803172/395c7a0b-effb-4df0-95e8-6560fb2b2c75">
